src/lib.rs

```diff
@@ -1,744 +1,345 @@
-    IRFilterMinMagMipPoint = 0,
-    IRFilterMinMagPointMipLinear = 0x1,
-    IRFilterMinPointMagLinearMipPoint = 0x4,
-    IRFilterMinPointMagMipLinear = 0x5,
-    IRFilterMinLinearMagMipPoint = 0x10,
-    IRFilterMinLinearMagPointMipLinear = 0x11,
-    IRFilterMinMagLinearMipPoint = 0x14,
-    IRFilterMinMagMipLinear = 0x15,
-    IRFilterAnisotropic = 0x55,
-    IRFilterComparisonMinMagMipPoint = 0x80,
-    IRFilterComparisonMinMagPointMipLinear = 0x81,
-    IRFilterComparisonMinPointMagLinearMipPoint = 0x84,
-    IRFilterComparisonMinPointMagMipLinear = 0x85,
-    IRFilterComparisonMinLinearMagMipPoint = 0x90,
-    IRFilterComparisonMinLinearMagPointMipLinear = 0x91,
-    IRFilterComparisonMinMagLinearMipPoint = 0x94,
-    IRFilterComparisonMinMagMipLinear = 0x95,
-    IRFilterComparisonAnisotropic = 0xd5,
-    IRFilterMinimumMinMagMipPoint = 0x100,
-    IRFilterMinimumMinMagPointMipLinear = 0x101,
-    IRFilterMinimumMinPointMagLinearMipPoint = 0x104,
-    IRFilterMinimumMinPointMagMipLinear = 0x105,
-    IRFilterMinimumMinLinearMagMipPoint = 0x110,
-    IRFilterMinimumMinLinearMagPointMipLinear = 0x111,
-    IRFilterMinimumMinMagLinearMipPoint = 0x114,
-    IRFilterMinimumMinMagMipLinear = 0x115,
-    IRFilterMinimumAnisotropic = 0x155,
-    IRFilterMaximumMinMagMipPoint = 0x180,
-    IRFilterMaximumMinMagPointMipLinear = 0x181,
-    IRFilterMaximumMinPointMagLinearMipPoint = 0x184,
-    IRFilterMaximumMinPointMagMipLinear = 0x185,
-    IRFilterMaximumMinLinearMagMipPoint = 0x190,
-    IRFilterMaximumMinLinearMagPointMipLinear = 0x191,
-    IRFilterMaximumMinMagLinearMipPoint = 0x194,
-    IRFilterMaximumMinMagMipLinear = 0x195,
-    IRFilterMaximumAnisotropic = 0x1d5,
-}
-
-#[repr(u32)]
-#[derive(Copy, Clone)]
-pub enum IRTextureAddressMode {
-    IRTextureAddressModeWrap = 1,
-    IRTextureAddressModeMirror = 2,
-    IRTextureAddressModeClamp = 3,
-    IRTextureAddressModeBorder = 4,
-    IRTextureAddressModeMirrorOnce = 5,
-}
-
-#[repr(u32)]
-#[derive(Copy, Clone)]
-pub enum IRComparisonFunction {
-    IRComparisonFunctionNever = 1,
-    IRComparisonFunctionLess = 2,
-    IRComparisonFunctionEqual = 3,
-    IRComparisonFunctionLessEqual = 4,
-    IRComparisonFunctionGreater = 5,
-    IRComparisonFunctionNotEqual = 6,
-    IRComparisonFunctionGreaterEqual = 7,
-    IRComparisonFunctionAlways = 8,
+impl Drop for IRRootSignature {
+    fn drop(&mut self) {
+        unsafe { self.funcs.IRRootSignatureDestroy(self.me) }
+    }
 }
 
-#[repr(u32)]
-#[derive(Copy, Clone)]
-pub enum IRStaticBorderColor {
-    IRStaticBorderColorTransparentBlack = 0,
-    IRStaticBorderColorOpaqueBlack = 1,
-    IRStaticBorderColorOpaqueWhite = 2,
-}
+impl IRRootSignature {
+    pub fn create_from_descriptor(
+        compiler: &IRCompiler,
+        desc: &IRVersionedRootSignatureDescriptor,
+    ) -> Result<IRRootSignature, Box<dyn std::error::Error>> {
+        unsafe {
+            let mut error: *mut IRError = std::ptr::null_mut::<IRError>();
```

Comment:
`MaybeUninit`?

Reply:
I'm not sure what you mean

Reply:
```result
let mut error = MaybeUninit::uninit();

function...(error.as_mut_ptr());

let error = unsafe { error.assume_init() };

if error.is_none() ...
```

Reply:
Let's postpone the error changes, but we can at least fully simplify the type here.